api/storage_service.cc

```diff
@@ -900,6 +902,129 @@ void set_storage_service(http_context& ctx, routes& r) {
             return make_ready_future<json::json_return_type>(map_to_key_value(std::move(status), res));
         });
     });
+
+    ss::sstable_info.set(r, [&ctx] (std::unique_ptr<request> req) {
+        auto ks = api::req_param<sstring>(*req, "keyspace", {}).value;
+        auto cf = api::req_param<sstring>(*req, "table", {}).value;
+
+        using table_sstables_list = std::vector<ss::table_sstables>;
```

Comment:
how big can this vector get?
There are two potential concerns with the API, stalls and large allocations.
I'm not sure it is an issue, but if the vector is big it can cause large allocation and if the structures are big, streaming it to JSON can cause a large allocation.
If it's not an issue we can ignore it

Reply:
It can absolutely get big if you ask for all sstables in the system. But "big" here is probably the order of 100/1000 entries, or whatever number of sstables in flight a node might have.

Reply:
Using a list will help with large alloc, but json serialization is another matter. Should that not be streaming?

Reply:
It all depends on the size, using stream (see compaction history for example) makes the code a bit more complex but is safer.
If it can grow big, you can use: `stream_range_as_array` (you can see examples in the api/storage_servie.cc file)

Reply:
Right. I've modified the final transform to use the streaming version.